manifests/init.pp

```diff
@@ -31,10 +31,10 @@
   Enum['mlocate','plocate']                   $locate,
   Optional[Array[String[1]]]                  $package_names = undef,
   Boolean                                     $ensure = true,
-  Array[String[1]]                            $prunefs = [],
+  Optional[Array[String[1]]]                  $prunefs = undef,
   Boolean                                     $prune_bind_mounts = true,
-  Array[Stdlib::Unixpath]                     $prunepaths = [],
-  Array[String[1]]                            $prunenames = [],
+  Optional[Array[Stdlib::Unixpath,1]]         $prunepaths = undef,
+  Optional[Array[String[1]]]                  $prunenames = undef,
```

Comment:
This does not look right: some OS have a value for these in Hiera so one cannot set them to `undef`.

Also, because we have values set by Hiera, the documentation is misleading. IMHO the default values should be here and not in Hiera (see my above comment), and if the data is in Hiera, there should be no default value here.

Reply:
Yes it's true only one OS for one parameter is undef with no hiera settings.

I like the defaults being correct to the OS.

Should we have a `defaults = true||false`

The not being to set something explicitly to undef except in hiera has always been difficult
to handle.

Oh and THANKS for the comments - much appreciated.